packages/frontend/src/components/SchedulerModals/SchedulerModalBase/SchedulerForm.tsx

```diff
@@ -221,9 +225,51 @@ const SchedulerForm: FC<{
         [slackChannelsQuery.data],
     );
     const health = useHealth();
-
+    const [openPicker] = useDrivePicker();
     const isAddSlackDisabled = disabled || slackState !== SlackStates.SUCCESS;
     const isAddEmailDisabled = disabled || !health.data?.hasEmailClient;
+    const hasGoogleDrive =
+        health.data?.auth.google.oauth2ClientId !== undefined &&
+        health.data?.auth.google.googleDriveApiKey !== undefined;
+    const { data: gdriveAuth, refetch } = useGdriveAccessToken();
+
+    const handleOpenPicker = useCallback(
+        (callback) => {
+            if (
+                !health.data?.auth.google.oauth2ClientId ||
+                !health.data.auth.google.googleDriveApiKey
+            )
+                return;
+
+            if (gdriveAuth === undefined) {
```

Comment:
auth flow in the frontend. If not authenticated, it will open a popup and `poll` to get the new `accessToken` once it is authenticated.